src/kirby/components/modal/services/modal.helper.ts

```diff
@@ -89,4 +149,40 @@ export class ModalHelper {
       }, 50);
     }
   }
+
+  private mergeDefaultConfig(config: ModalConfig): ModalConfig {
```

Comment:
Consider simplifying the mergeDefaultConfig to something like this:

```typescript
private mergeDefaultConfig(config: ModalConfig): ModalConfig {
    let defaults;

    if (config.flavor === 'drawer') {
      defaults = {
        enterDuration: KirbyAnimation.Duration.LONG,
        leaveDuration: KirbyAnimation.Duration.LONG,
        easingIn: KirbyAnimation.Easing.ENTER,
        easingOut: KirbyAnimation.Easing.EXIT,
      };
    } else {
      defaults = {
        flavor: 'modal',
        dim: ModalConfigHelper.defaultDim,
        enterDuration: KirbyAnimation.Duration.SHORT,
        leaveDuration: KirbyAnimation.Duration.SHORT,
        easingIn: KirbyAnimation.Easing.STATIC,
        easingOut: KirbyAnimation.Easing.STATIC,
      };
    }

    return { ...defaults, ...config };
  }
```

Reply:
Very much better, fixed